### PR TITLE
LibraryDictionarySource fix possible leak

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Dictionaries/LibraryDictionarySource.cpp [hunk 1]
```diff
@@ -193,15 +192,20 @@ BlockInputStreamPtr LibraryDictionarySource::loadAll()
...
     }
+
     void * data_ptr = nullptr;
 
     /// Get function pointer before dataNew call because library->get may throw.
     auto func_load_all
         = library->get<void * (*)(decltype(data_ptr), decltype(&settings->strings), decltype(&columns))>("ClickHouseDictionary_v3_loadAll");
+
     data_ptr = library->get<decltype(data_ptr) (*)(decltype(lib_data))>("ClickHouseDictionary_v3_dataNew")(lib_data);
+    SCOPE_EXIT(library->get<void (*)(decltype(lib_data), decltype(data_ptr))>("ClickHouseDictionary_v3_dataDelete")(lib_data, data_ptr));
```
Comment (kitaisreal) [inline src/Dictionaries/LibraryDictionarySource.cpp:203]: @alexey-milovidov, @alesapin  I assume if dataToBlock function throws exception before we does not call deleter for data that we get from shared library.

**Review by alesapin — APPROVED** [review]
LGTM

**Review by alexey-milovidov — APPROVED** [review]

File: src/Common/SharedLibrary.cpp [hunk 2]
```diff
@@ -13,11 +13,11 @@ namespace ErrorCodes
     extern const int CANNOT_DLSYM;
 }
 
-SharedLibrary::SharedLibrary(const std::string & path, int flags)
+SharedLibrary::SharedLibrary(const std::string_view & path, int flags)
```
Comment (alexey-milovidov) [inline src/Common/SharedLibrary.cpp:16]: Pass by value?

Comment (kitaisreal) [reply src/Common/SharedLibrary.cpp]: Fixed.

**Pull request merged**